### PR TITLE
Convert partially-sigged test to testdata test

### PR DESCRIPTION
**Pull request by jez**
Review by commit.

### Motivation

This test was previously disabled. We don't really need it to be tested,
because we can test it another way.

### Test plan

See included automated tests.

### CODE REVIEW

File: test/testdata/rbi/with_without__3.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# typed: true
+
+T.reveal_type(MyGem.method_thats_part_of_my_public_api) # error: Revealed type: `NilClass`
```
Comment (elliottt) [inline test/testdata/rbi/with_without__3.rb]: For this file and the two rbis, is it expected that they're all present for one tests, or are they each different tests?

Comment (jez) [reply test/testdata/rbi/with_without__3.rb]: the `foo__N.rb` syntax is a thing the test corpus runner already was aware of prior to this change. it's a way to make a single bazel test target that contains multiple ruby files.

in this change, I added the ability to intermix `foo__N.rbi` files with the `.rb` files.

Comment (jez) [reply test/testdata/rbi/with_without__3.rb]: e.g., this existing test:

https://github.com/sorbet/sorbet/blob/d925413dff03c1b58d390b8d73f53fe13da8fdc8/test/testdata/infer/aliases__1.rb
https://github.com/sorbet/sorbet/blob/d925413dff03c1b58d390b8d73f53fe13da8fdc8/test/testdata/infer/aliases__2.rb

is one bazel target for a test that contains code in two files.

Comment (elliottt) [reply test/testdata/rbi/with_without__3.rb]: Ah, nice!

**Pull request merged**